lib/block-parser.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+if (!class_exists('Gutenberg_Block_Parser_State', false)):
+
+class Gutenberg_Block_Parser_State {
+	public $block_stack;
+}
+
+endif;
+
+if (!class_exists('Gutenberg_Block_Parser', false)):
+
+class Gutenberg_Block_Parser {
+	const BLOCK_COMMENT_OPEN  = '(^<!--)';
+	const BLOCK_COMMENT_CLOSE = '(^/?-->)';
+	const BLOCK_NAME          = '(^[[:alpha:]](?:[[:alnum:]]|/[[:alnum:]])*)i';
```

Comment:
We still need to be able to safely parse blocks from plugins that have been disabled or uninstalled. This could work by including them within larger chunks of content, but I think it is better to show them as an independent but "unrecognized" block.  See #1735.

I also don't like the idea of coupling the parser to the list of registered blocks.